### PR TITLE
distsql: prework for analyze pushdown

### PR DESCRIPTION
**Pull request by lamxTyler**
This is part of pre work for analyzing push down. This PR implements a distsql request for analyze, and a special NextRaw for the use of analyze.
PTAL @coocood  @hanfei1991  @winoros

### CODE REVIEW

**Comment by coocood** [comment]
Have you considered to reuse the old iterator by modifying it?

**Comment by coocood** [comment]
We can add `NextRaw` on `SelectResult`, and change `resultWithErr.result` type to `[]byte`.
unmarshal to `PartialResult` only on `Next` method, return the raw []byte on `NextRaw`.

File: distsql/analyze.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	var err error
+	kvReq.Data, err = req.Marshal()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	resp := client.Send(ctx, kvReq)
+	if resp == nil {
+		err = errors.New("client returns nil response")
+		return nil, errors.Trace(err)
```
Comment (zz-jason) [inline distsql/analyze.go]: just `return nil, errors.New("client returns nil response")`, no need to `Trace`

**Comment by coocood** [comment]
LGTM

**Comment by zimulala** [comment]
LGTM

**Comment by lamxTyler** [comment]
@zimulala PTAL again, Thanks.

**Pull request merged**